materialize-elasticsearch/type_mapping.go

```diff
@@ -145,30 +160,20 @@ func (constrainter) NewConstraints(p *pf.Projection, deltaUpdates bool) *pm.Resp
 	return &constraint
 }
 
-func (constrainter) Compatible(existing boilerplate.EndpointField, proposed *pf.Projection, _ json.RawMessage) (bool, error) {
-	prop := propForProjection(proposed).Type
-
-	if strings.EqualFold(existing.Type, string(elasticTypeText)) {
-		// Allow any of the "formatted string" types to be materialized into an existing text
```

Comment:
I'm removing this compatibility since it is not consistent with other materializations (ex: SQL) and I think it was an oversight in the first place. No ES materializations that I'm aware of are relying on this behavior right now.

Reply:
So our position now is that we don't want customers to bring their own table / index if it doesn't conform to the exact schema that we have for the collection? In case of keyword we are allowing this if they tell us about it in their field selection. So for further compatibility between these types we would need similar configurations in field selection?

Reply:
To the first question, yes, and this is really because we want re-backfills to be as unsurprising as possible. If a pre-existing index had a `text` mapping that was being materialized as a `format: xxxxx` string collection field, a re-backfill would re-create that mapping as the formatted type rather than the original `text` mapping, which could be surprising. Similar behavior exists in SQL materializations, where pre-existing text columns could by all rights be materialized into from `type: string, format: xxxxx` fields, but we prohibit that so re-backfills don't alter columns that were previously validated as compatible.

For the second question, we could perhaps formalize a field configuration that would ignore the format of strings, although I'm not sure if that's the right thing to do right now. This the `keyword` vs. `text` field config option is controlling behavior in the materialized system that we can't otherwise represent in the source collection schema via a `format` annotation.